python/feathub/common/configuration.py

```diff
@@ -0,0 +1,194 @@
+#  Copyright 2022 The Feathub Authors
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+from enum import Enum
+from typing import TypeVar, Generic, Optional, Dict, Type, Any, List
+
+import pandas as pd
+
+from feathub.common.exceptions import FeathubConfigurationException
+
+T = TypeVar("T")
+
+
+# TODO: Generate documentation of all the defined configuration automatically.
+class FeathubConfiguration:
+    """
+    FeathubConfiguration holds all the configurations.
+    """
+
+    def __init__(self) -> None:
+        self.config: Dict[str, Any] = {}
+
+    @staticmethod
+    def from_dict(config_dict: Dict) -> "FeathubConfiguration":
+        config = FeathubConfiguration()
+        json_normalized = pd.json_normalize(config_dict).to_dict(orient="records")
+        if len(json_normalized) >= 1:
+            [config.config] = json_normalized
+        else:
+            config.config = {}
+        return config
+
+    def get(self, config_option: "ConfigOption[T]") -> T:
```

Comment:
It seems better to allow users to get config value by providing a string that uniquely identified the config.